src/js/angular/import/controllers.js

```diff
@@ -695,20 +793,43 @@ importViewModule.controller('UploadCtrl', ['$scope', 'toastr', '$controller', '$
             }
         );
         $scope.savedSettings = _.mapKeys(_.filter($scope.files, 'parserSettings'), 'name');
-        _.each($scope.currentFiles, function (file) {
-            $scope.updateImport(file.name);
-        });
+        if (isFileListInitialized) {
+            _.each($scope.currentFiles, function (file) {
+                $scope.updateImport(file.name);
+            });
+        }
     };
 
     const removeAllListeners = () => {
         subscriptions.forEach((subscription) => subscription());
     };
 
+    /**
+     * Builds a registry of filenames mapped to numerical indices.
+     * The registry is in format <code>{filename: index}</code>.
+     * @param {[object]} files - list of files to be used when building the registry.
+     */
+    const buildPrefixesRegistry = (files) => {
+        files.forEach((file) => {
+            // file name is in format 123-file-name.txt
+            // split the number from the file name
+            const separatorIndex = file.name.indexOf('-');
+            let index = file.name.substr(0, separatorIndex);
```

Comment:
Maybe the separator is not correct. Try to import an RDF text snippet, its name not start with number and "-".